lib/shopify_app/login_protection.rb

```diff
@@ -26,6 +26,7 @@ def shop_session
 
     def login_again_if_different_shop
       if shop_session && params[:shop] && params[:shop].is_a?(String) && shop_session.url != params[:shop]
+        session[:shopify] = nil
```

Comment:
We also set `:shopify_domain` in the session here: https://github.com/Shopify/shopify_app/blob/dd7feffd03ab047cfbf4fad4187f11ca2aa5cec3/lib/shopify_app/sessions_controller.rb#L18 

Might be worth doing a `reset_session` instead?